test/transports/routing.test.ts

```diff
@@ -456,6 +455,246 @@ test.serial('RoutingTransport can route to SSE transport', async (t) => {
+    name: 'TEST',
+    defaultEndpoint: 'price',
+    endpoints: [endpoint],
+    rateLimiting: {
+      tiers: {
+        default: {
+          rateLimit1s: 5,
+        },
+      },
+    },
+    envDefaultOverrides: {
+      LOG_LEVEL: 'debug',
+      METRICS_ENABLED: false,
+      CACHE_POLLING_SLEEP_MS: 10,
+      CACHE_POLLING_MAX_RETRIES: 0,
+    },
+  })
+
+  const api = await expose(adapter)
+  const address = `http://localhost:${(api?.server.address() as AddressInfo)?.port}`
+  const price = 1500
+
+  nock(restUrl)
+    .post('/price', {
+      pairs: [
+        {
+          base: from,
+          quote: to,
+        },
+      ],
+    })
+    .reply(200, {
+      prices: [
+        {
+          pair: `${from}/${to}`,
+          price,
+        },
+      ],
+    })
+    .persist()
+
+  const makeRequest = () =>
+    axios.post(address, {
+      data: {
+        from,
+        to,
+        transport: 'BATCH',
+      },
+    })
+
+  const error: AxiosError | undefined = await t.throwsAsync(makeRequest)
+
+  t.is(error?.response?.status, 504)
+  const internalTransport = transports['batch'] as MockHttpTransport
   t.assert(internalTransport.backgroundExecuteCalls > 0)
 })
+
+test.serial('custom router returns invalid transport and request fails', async (t) => {
+  const { from, to } = t.context
+  const transport = new RoutingTransport<BaseEndpointTypes>(transports, {
+    customRouter: () => {
+      return 'qweqwe'
+    },
+  })
+  const endpoint = new AdapterEndpoint<BaseEndpointTypes>({
+    inputParameters,
+    name: 'price', // /price
+    transport,
+  })
+
+  const adapter = new Adapter<typeof CustomSettings>({
+    name: 'TEST',
+    defaultEndpoint: 'price',
+    endpoints: [endpoint],
+    rateLimiting: {
+      tiers: {
+        default: {
+          rateLimit1s: 5,
+        },
+      },
+    },
+    envDefaultOverrides: {
+      LOG_LEVEL: 'debug',
+      METRICS_ENABLED: false,
+      CACHE_POLLING_SLEEP_MS: 10,
+      CACHE_POLLING_MAX_RETRIES: 0,
+    },
+  })
+
+  const api = await expose(adapter)
+  const address = `http://localhost:${(api?.server.address() as AddressInfo)?.port}`
+  const price = 1500
+
+  nock(restUrl)
+    .post('/price', {
+      pairs: [
+        {
+          base: from,
+          quote: to,
+        },
+      ],
+    })
+    .reply(200, {
+      prices: [
+        {
+          pair: `${from}/${to}`,
+          price,
+        },
+      ],
+    })
+    .persist()
+
+  const makeRequest = () =>
+    axios.post(address, {
+      data: {
+        from,
+        to,
+        transport: 'BATCH',
+      },
+    })
+
+  const error: AxiosError | undefined = await t.throwsAsync(makeRequest)
+
+  t.is(error?.response?.status, 400)
+  t.is(
+    (error?.response?.data as { error: { message: string } }).error?.message,
+    'No transport found for qweqwe',
+  )
+})
+
+test.serial('missing transport in input params with no default fails request', async (t) => {
```

Comment:
Not sure if I missed it but is there a test for missing transport but a default specified?

Reply:
Oh sorry missing transport is probably the wrong phrasing. I meant if one isn't specified in the request but there's a default specified. Just another success scenario. Think all of the ones above have a transport specified.